src/app/system/manage-data-tables/create-data-table/create-data-table.component.ts

```diff
@@ -0,0 +1,175 @@
+  /** Column Type Data */
+  columnTypeData: any = ['String', 'Number', 'Decimal', 'Boolean', 'Date', 'DateTime', 'Text', 'Dropdown'];
+  /** Code Data */
+  codeData: any;
+
+  /**
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {SystemService} systemService System Service.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   */
+  constructor(private formBuilder: FormBuilder,
+    private systemService: SystemService,
+    private route: ActivatedRoute,
+    private router: Router) {
+    this.route.data.subscribe((data: { codes: any }) => {
+      this.codeData = data.codes;
+    });
+  }
+
+  /**
+   * Creates Data Table Form.
+   * Subscribes to the Column Type.
+   */
+  ngOnInit() {
+    this.createDataTableForm();
+    this.onColumnTypeChanges();
+  }
+
+  /**
+   * Creates the data table form.
+   */
+  createDataTableForm() {
+    this.dataTableForm = this.formBuilder.group({
+      'datatableName': ['', Validators.required],
+      'apptableName': ['', Validators.required],
+      'multiRow': [''],
+      'columns': this.formBuilder.array([this.createColumnForm()])
+    });
+  }
+
+  /**
+   * Creates the Column Form.
+   * @return {FormGroup} Column Form.
+   */
+  createColumnForm(): FormGroup {
+    return this.formBuilder.group({
+      'name': ['', Validators.required],
+      'type': ['', Validators.required],
+      'mandatory': [''],
+      'length': [{ value: '', disabled: true }, Validators.required],
+      'code': [{ value: '', disabled: true }, Validators.required]
+    });
+  }
+
+  /**
+   * Gets the columns form array.
+   * @returns {FormArray} Columns form array.
+   */
+  get columns(): FormArray {
+    return this.dataTableForm.get('columns') as FormArray;
+  }
+
+  /**
+   * Adds the column form to given column form array.
+   * @param {FormArray} columnFormArray Given column form array.
+   * Subscribes to the Column Type for the updated Form Array.
+   */
+  addColumn(columnFormArray: FormArray) {
+    columnFormArray.push(this.createColumnForm());
+    this.onColumnTypeChanges();
+  }
+
+  /**
+   * Removes the column form from given column form array at given index.
+   * @param {FormArray} columnFormArray Given column form array.
+   * @param {number} index Array index from where column form needs to be removed.
+   * Subscribes to the Column Type for the updated Form Array.
+   */
+  removeColumn(columnFormArray: FormArray, index: number) {
+    columnFormArray.removeAt(index);
+    this.onColumnTypeChanges();
+  }
+
+  /**
+   * Enables Length Text Field and disables Code Text Field when Column Type is 'String'.
+   * Enables Code Text Field and disables Length Text Field when Column Type is 'Dropdown'.
+   * Disables both Length and Code Text Field when Column Type is neither 'String' nor 'Dropdown'.
+   */
+  onColumnTypeChanges() {
```

Comment:
You are calling this function again and again for the whole array whenever a single row is added or removed. Update this so that it is only called once when a row is created and for that row only not the whole array.